test/metabase/query_processor/util/nest_query_test.clj

```diff
@@ -762,3 +762,24 @@
                     (-> query'
                         nest-query/nest-expressions
                         readable-query)))))))))
+
+(deftest ^:parallel nest-expressions-ignores-temporal-units-from-joined-fields
+  (testing "clear temporal units from joined fields #48058"
+    (driver/with-driver :h2
+      (qp.store/with-metadata-provider meta/metadata-provider
+        (is (empty?
+             (->> (lib.tu.macros/mbql-query orders
+                    {:expressions {"double_total" [:* $total 2]}
+                     :breakout    [!hour-of-day.people.created-at
+                                   [:expression "double_total"]]
+                     :aggregation [[:count]]
+                     :joins [{:source-table $$people
+                              :alias        "p"
+                              :condition    [:= $user-id &p.people.id]}]})
+                  qp.preprocess/preprocess
+                  add/add-alias-info
+                  nest-expressions
+                  :source-query
+                  :fields
+                  (map #(% 2))
+                  (filter :temporal-unit))))))))
```

Comment:
This feels like squinting too hard, stripping everything away from the test output. We tend to leverage the `=?` fuzzy matcher to get a more solid picture of the expectation.

Let's try something more like `(is (=? {:source-query {:fields [[:field (mt/id :some :field) {:temporal-unit (symbol "nil #_\"key is not present.\"")}] ...]}} ...))`.

Reply:
sure